front/src/common/api/osrdEditoastApi.ts

```diff
@@ -86,6 +86,9 @@ const osrdEditoastApi = generatedEditoastApi.enhanceEndpoints({
     deleteV2ProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId: {
       invalidatesTags: ['scenariosv2', 'studies', 'projects'],
     },
+    postV2TrainScheduleProjectPath: {
+      invalidatesTags: [],
+    }
```

Comment:
I don't think we should remove the tag in this endpoint `invalidatesTags` but rather remove for now the `trainSchedulev2` in the `postV2TrainScheduleSimulationSummary` `providesTags` so that any of the other trainschedule mutation endpoints don't trigger summary.

The workflow in trainschedule v2 when adding/editing a train : `timetablev2` tag is invalidated, we get the new trainIds required to call `postV2TrainScheduleSimulationSummary` and `postV2TrainSchedule` that give us the train details to display in the timetable. So we don't really need to have any automate refetching other than the timetable one.

We plan one day to use more granular control to invalidate cache with rtk query (https://redux-toolkit.js.org/rtk-query/usage/automated-refetching#invalidating-cache-data)

Reply:
trk-query doesn't auto-generate any `providesTags` for mutations. It only generates `invalidatesTags`. See:

https://github.com/OpenRailAssociation/osrd/blob/dev/front/src/common/api/generatedEditoastApi.ts#L928-L938

Reply:
TBH it's a bit weird to use a mutation for this request. Would be nicer to add it here so that it's generated as a query instead:

https://github.com/OpenRailAssociation/osrd/blob/de8ada893e20f4b16cee8f34f7b60e7f53d7625d/front/src/config/openapi-editoast-config.ts#L13

But that's kinda unrelated to the original bug.

Reply:
why does postV2TrainScheduleProjectPath need to invalidate that tag in the first place ?

Reply:
> TBH it's a bit weird to use a mutation for this request. Would be nicer to add it here so that it's generated as a query instead:
> 
> https://github.com/OpenRailAssociation/osrd/blob/de8ada893e20f4b16cee8f34f7b60e7f53d7625d/front/src/config/openapi-editoast-config.ts#L13
> 
> But that's kinda unrelated to the original bug.

I think it considers all POST/PUT as mutations. `train_schedule/project_path` is creating or modifying something in DB ?

Reply:
> I think it considers all POST/PUT as mutations.

Yup, that's the default behavior (can be changed by editing the referenced file).

> train_schedule/project_path is creating or modifying something in DB ?

I don't believe it does? (Correct me if I'm wrong)

Reply:
train_schedule/project_path is confirmed not to be modifying anything, so it has no reason to invalidate a tag that we would need to refetch.

Reply:
Agreed for the projectpath but I still think we should also remove the trainSchedulev2 tag from summary endpoint because we will have the same issue when post/put a trainSchedule

Reply:
Hm, what is your concern with `postV2TrainScheduleSimulationSummary`? When we create/edit/delete a train schedule, everything with the `train_schedulev2` tag gets invalidated, so the summary is pruned from the cache, I think?

Reply:
It might have been lost in the conversation, but in my first comment, I described the workflow used in the app in the timetable section : 
![Capture d’écran 2024-07-12 à 15 04 27](https://github.com/user-attachments/assets/232c2a36-d151-4fd0-b68b-ead1f7a09699)

To summarize, we will refetch summary anyway because add/put trainschedule invalidates timetable which cascades into summary.

Reply:
I’m still a bit puzzled, if you modify the train start time, the result will be the same, no need to recall summary. but  if you modify the train path,  it won’t, so summary needs to be recalled.